publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/identifiers/HandleIdentifierEventHandler.java

```diff
@@ -0,0 +1,175 @@
+             S3Driver.defaultS3Client().build(),
+             new Environment(),
+             HttpClient.newBuilder().build(),
+             SecretsReader.defaultSecretsManagerClient());
+    }
+
+    protected HandleIdentifierEventHandler(ResourceService resourceService,
+                                           S3Client s3Client,
+                                           Environment environment,
+                                           HttpClient httpClient,
+                                           SecretsManagerClient secretsManagerClient) {
+        String apiDomain = environment.readEnv("API_DOMAIN");
+        String handleBasePath = environment.readEnv("HANDLE_BASE_PATH");
+        this.backendClientSecretName = environment.readEnv("BACKEND_CLIENT_SECRET_NAME");
+        this.backendClientAuthUrl = environment.readEnv("BACKEND_CLIENT_AUTH_URL");
+        this.frontendDomain = environment.readEnv("NVA_FRONTEND_DOMAIN");
+        this.resourceService = resourceService;
+        this.s3Driver = new S3Driver(s3Client, PublicationEventsConfig.EVENTS_BUCKET);
+        this.secretsManagerClient = new SecretsReader(secretsManagerClient);
+        AuthorizedBackendClient authorizedBackendClient = AuthorizedBackendClient.prepareWithCognitoCredentials(
+            httpClient,
+            fetchCredentials());
+        this.handleService = new HandleService(authorizedBackendClient, apiDomain, handleBasePath);
+    }
+
+    @Override
+    public Void handleRequest(SQSEvent sqsEvent, Context context) {
+        sqsEvent.getRecords()
+            .stream()
+            .map(sqs -> {
+                logger.info("Processing sqsEvent: {}", sqs.getBody());
+                try {
+                    return JsonUtils.dtoObjectMapper
+                               .readValue(sqs.getBody(),
+                                          new TypeReference<AwsEventBridgeEvent<AwsEventBridgeDetail<EventReference>>>() {
+                                          })
+                               .getDetail()
+                               .getResponsePayload();
+                } catch (JsonProcessingException e) {
+                    throw new RuntimeException(e);
+                }
+            })
+            .forEach(this::processInputPayload);
+        return null;
+    }
+
+    private CognitoCredentials fetchCredentials() {
+        var credentials = secretsManagerClient.fetchClassSecret(backendClientSecretName,
+                                                                BackendClientCredentials.class);
+        var uri = UriWrapper.fromHost(backendClientAuthUrl).getUri();
+        return new CognitoCredentials(credentials::getId, credentials::getSecret, uri);
+    }
+
+    protected Void processInputPayload(EventReference input) {
+        var eventBlob = s3Driver.readEvent(input.getUri());
+
+        if (RESOURCE_UPDATE_EVENT_TOPIC.equals(input.getTopic())) {
+            var resourceUpdate = parseResourceUpdateInput(eventBlob);
```

Comment:
You dont want to listen for this event directly from event bus, right?

Reply:
no, I fear that the db could be a bottle neck or have down time, so would like a sqs in between so that nothing is lost in transit